src/components/MainCarousel.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react'
+import ChevronLeft from 'icons/ChevronLeft'
+import ChevronRight from 'icons/ChevronRight'
+import classnames, {
+  alignItems,
+  backgroundColor,
+  borderRadius,
+  display,
+  flex,
+  flexDirection,
+  fontSize,
+  gap,
+  height,
+  justifyContent,
+  margin,
+  opacity,
+  overflow,
+  scrollSnap,
+  textAlign,
+  textColor,
+  width,
+} from 'classnames/tailwind'
+
+const wrapper = classnames(
+  display('flex'),
+  flexDirection('flex-col'),
+  gap('gap-6')
+)
+const container = classnames(
+  width('w-64', 'xs:w-80', 'md:w-104'),
+  display('inline-flex'),
+  overflow('overflow-scroll'),
+  scrollSnap('snap-x', 'snap-mandatory')
+)
+const slide = classnames(
+  width('w-full'),
+  display('flex'),
+  flex('flex-none'),
```

Comment:
You sure we need it?

Reply:
Yes, it can't shrink into the container.